File: Assignment9/program4.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program4.c
//  Description : Accept a Integer and counts 4 in it
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 17/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

////////////////////////////////////////////////////////////////
// 
//  Function Name : CountFour
//  Description   : Returns count of 4 for input integer
//  Input         : Integer
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

int CountFour(int iNo)
{
    int iDigit = 0;
    int iFreq = 0;
    
    if(iNo < 0)
    {
        iNo = -iNo;
    }
    
    while (iNo != 0)
    {
        iDigit = iNo % 10;
        
        if(iDigit == 4)
        {
            iFreq  = iFreq + 1;
        }
        iNo = iNo / 10;
    }

    return iFreq ;
}

int main()
{
    int iValue =0;
    int iRet = 0;
 
    printf("enter number");
    scanf("%d",&iValue);

    iRet= CountFour(iValue);
 
    printf("%d",iRet);
 
    return 0;
}

File: Assignment26/program4.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program4.java
//  Description : Accept a String and a Character and returns freqency index of Last occurance of that character in String
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 29/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Display
//  Function Name : LastChar
//  Description   : Inputs String and a Charater and returns freqency index of Last occurance of that character in String
//  Input         : String
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

class Display
{
    public int FirstChar(char[] chVal, int iSize, char chSearch)
    {
        int iCnt = 0;
        int iCount = -1;

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            if(chVal[iCnt] == chSearch)
            {
                iCount = iCnt;
            }
        }
        return iCount;
    }
}
public class program4
{
    public static void main(String args[])
    {
        int iCnt = 0;
        int iLength = 0;
        char[] cValue = null;
        String sValue = null;
        char cSearch = '\0';
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter String : ");
        sValue = sobj.nextLine();

        System.out.println("Enter the character you wanna Search in String : ");
        cSearch = sobj.next().charAt(0);

        iLength = sValue.length();

        cValue = new char[iLength];

        for(iCnt = 0; iCnt < iLength; iCnt++)
        {
            cValue[iCnt] = sValue.charAt(iCnt);
        }

        Display dobj = new Display();

        iRet = dobj.FirstChar(cValue, iLength, cSearch);

        System.out.println("Frequency Index of First Occurance of "+cSearch+" in our String is : "+iRet);
        
        sobj.close();
    }
}